parserator_web/views.py

```diff
@@ -8,17 +9,55 @@
 
 class Home(TemplateView):
     template_name = 'parserator_web/index.html'
+    print("check")
+
 
 
 class AddressParse(APIView):
     renderer_classes = [JSONRenderer]
-
+   
     def get(self, request):
         # TODO: Flesh out this method to parse an address string using the
         # parse() method and return the parsed components to the frontend.
-        return Response({})
+
+        #Store input string received from user input.
+        input_string = request.GET['address']
+
+        #Runs the parse method. Will return an array of usable components if the input was valid or will return a repeat==True value if the the format was repeated.
+        parsed = AddressParse.parse(self, input_string)
+
+        #Checks to see if input was valid. If so, returns the API response of components that will be used to render table on the user's end.
+        if parsed[0] == bool(False):
```

Comment:
this if/else block basically replicates the try/except block you have in the parse method. what about moving the try/except block up to here?

Reply:
Great idea! Thanks!